### DISCUSSION TITLE
gfx

### DISCUSSION THREAD

**Discussion by Smokiiee** [kibook/pmms#17] [General]
hi :) nice script:D i like it. But how do you know how to get a gfx file or make one?

**Comment by kibook** [comment]
The gfx file format is part of Scaleform, the system Rockstar used to make the UI in GTA V. It's based on Adobe Flash:

https://en.wikipedia.org/wiki/Scaleform_GFx

A .gfx file is essentially just a .swf file, so to edit or create one you'd use Adobe Flash or some other tool that works with .swf files, and then use another tool to convert it between SWF and GFX, or do so by hand in a hex editor since it's just a small part of the header that needs to be changed:

https://help.autodesk.com/view/SCLFRM/ENU/?guid=__scaleform_help_gfxexport_reference_faqs_html
http://wiki.tesnexus.com/index.php/Flash_SWF_Editing
https://docs.cryengine.com/display/CEMANUAL/2+-+Flash+and+Gfx
https://sourceforge.net/projects/btek-kingfish/files/CryENGINE%20Modded/Tools/GFxExport/

The pmms_texture_renderer.gfx is just a simple GFX object that can have a single texture applied to it. I didn't create it myself, I used this one provided by thers on the Cfx forums:

https://forum.cfx.re/t/release-generic-dui-2d-3d-renderer/131208
